hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoCreateValidator.java

```diff
@@ -0,0 +1,102 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl.validators;
+
+import static com.hedera.hapi.node.base.ResponseCodeEnum.ALIAS_ALREADY_ASSIGNED;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.INVALID_ALIAS_KEY;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.KEY_REQUIRED;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.NOT_SUPPORTED;
+import static com.hedera.node.app.spi.workflows.HandleException.validateFalse;
+import static com.hedera.node.app.spi.workflows.HandleException.validateTrue;
+
+import com.hedera.hapi.node.token.CryptoCreateTransactionBody;
+import com.hedera.node.app.service.token.ReadableAccountStore;
+import com.hedera.node.app.spi.validation.AttributeValidator;
+import com.hedera.node.app.spi.workflows.HandleException;
+import com.hedera.node.config.data.CryptoCreateWithAliasConfig;
+import com.hedera.node.config.data.EntitiesConfig;
+import com.hedera.node.config.data.LedgerConfig;
+import com.hedera.node.config.data.TokensConfig;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Provides validation for token fields like token type,  token supply type, token symbol etc.,.
+ * It is used in pureChecks for token creation.
+ */
+@Singleton
+public class CryptoCreateValidator {
+    private static final int EVM_ADDRESS_SIZE = 20;
+
+    @Inject
+    public CryptoCreateValidator() { // Exists for injection
+    }
+
+    /**
+     * Validates Key Alias and EVM Address combinations.
+     *
+     * @param op    the crypto create transaction body
+     * @param attributeValidator   AttributeValidator
+     * @param config    CryptoCreateWithAliasConfig
+     * @param readableAccountStore    ReadableAccountStore
+     */
+    public void validateKeyAliasAndEvmAddressCombinations(
+            @NonNull final CryptoCreateTransactionBody op,
+            @NonNull final AttributeValidator attributeValidator,
+            @NonNull final CryptoCreateWithAliasConfig config,
+            @NonNull final ReadableAccountStore readableAccountStore) {
+        if (op.hasKey() && op.alias().equals(Bytes.EMPTY)) {
+            attributeValidator.validateKey(op.keyOrThrow());
+        } else if (!op.hasKey() && !op.alias().equals(Bytes.EMPTY)) {
+            validateTrue(config.enabled(), NOT_SUPPORTED);
+            validateFalse(config.enabled(), INVALID_ALIAS_KEY);
```

Comment:
Validating both true and false here of `config.enabled()` doesn't make sense to me. Will it ever succeed?

Reply:
One is validateTrue, another is validateFalse even though pass in the same config.enable. It's under an if statement. mono code makes more sense: dynamicProperties.isCryptoCreateWithAliasEnabled() ? INVALID_ALIAS_KEY : NOT_SUPPORTED. Here I tried to use validateTrue/False to follow the other pattern.

Reply:
Ah I see. In this case, may I suggest:
```java
final var responseCode = config.enabled() ? NOT_SUPPORTED : INVALID_ALIAS_KEY;
throw new HandleException(responseCode)
```
to throw the exception explicitly instead of using `validateTrue` or `validateFalse`

Reply:
Updated the code. Checking into validateTrue(), you will see the code is a little different from your suggested here.